tests/letstest/multitester.py

```diff
@@ -341,22 +338,28 @@ def create_client_instances(targetlist, security_group_id, subnet_id):
     return instances
 
 
-def test_client_process(inqueue, outqueue):
+def test_client_process(inqueue, outqueue, boulder_url):
     cur_proc = mp.current_process()
     for inreq in iter(inqueue.get, SENTINEL):
-        ii, target = inreq
+        ii, instance_id, target = inreq
+
+        # Each client process is given its own session due to the suggestion at
+        # https://boto3.amazonaws.com/v1/documentation/api/latest/guide/resources.html?highlight=multithreading#multithreading-multiprocessing.
+        aws_session = boto3.session.Session(profile_name=PROFILE)
+        ec2_client = aws_session.resource('ec2')
+        instance = ec2_client.Instance(id=instance_id)
```

Comment:
So if I understood well the logic here, these three lines are the core modification of this PR, and everything else is either to make theses lines possibles, or to adapt the logic from theses lines.

Am I right?

Reply:
Pretty much. A higher level summary of the changes in this PR which I probably should have given before is I created a new AWS session and EC2 resource in each client process, don't pass any `boto3` objects through the shared queues, and removed the global `boto3` state by putting it in a `main()` function. This last change caused a lot of other changes to have to be made.